arangod/Aql/OptimizerRulesCluster.cpp

```diff
@@ -439,6 +434,111 @@ bool substituteClusterSingleDocumentOperationsNoIndex(
   return modified;
 }
 
+bool substituteClusterMultipleDocumentInsertOperations(
+    Optimizer* opt, ExecutionPlan* plan, OptimizerRule const& rule) {
+  containers::SmallVector<ExecutionNode*, 8> nodes;
```

Comment:
This optimization rule triggers in cases such as 
``` 
 - `FOR doc IN @docs INSERT doc INTO ...` (where `@docs` is a bind parameter
    with an array of documents to be inserted),
  - `FOR doc IN [...] INSERT doc INTO ...` (where the FOR loop iterates over
    an array of input documents known at query compile time),
  - `LET docs = [...] FOR doc IN docs INSERT doc INTO ...` (where the documents
    set up by the LET are some static documents known at query compile time),
     doc = {_key: '123', value: 2};
INSERT @doc INTO ...
```
 when the insertion node is fed input from a node that is either a constant calculation node such as `FOR d IN [{value: 1}]` or an enumerate list. 
 This, for now, doesn't support RETURN statements or later modification. 
 What it does is optimize the number of HTTP requests by being greedy and inserting all at once instead of batching. 
 So if you insert 9k documents into a collection with 3 shards and a 3-node cluster, 3k documents are sent to each DB-Server with 3 http requests, one for each server
 
 it's not triggered when the INSERT statement comes after a node that is not a constant calculation or an enumerate list node, nor does it support RETURN statements or more than one INSERT in the same request or nested queries.